test/e2e/httpproxy/httpproxy_test.go

```diff
@@ -16,95 +16,169 @@
+
+	f.NamespacedTest("003-path-condition-match", testPathConditionMatch)
+
+	f.NamespacedTest("004-https-sni-enforcement", testHTTPSSNIEnforcement)
+
+	f.NamespacedTest("005-pod-restart", testPodRestart)
+
+	f.NamespacedTest("006-merge-slash", testMergeSlash)
+
+	f.NamespacedTest("007-client-cert-auth", testClientCertAuth)
+
+	f.NamespacedTest("008-tcproute-https-termination", testTCPRouteHTTPSTermination)
+
+	f.NamespacedTest("009-https-misdirected-request", testHTTPSMisdirectedRequest)
+
+	f.NamespacedTest("010-include-prefix-condition", testIncludePrefixCondition)
+
+	f.NamespacedTest("011-retry-policy-validation", testRetryPolicyValidation)
+
+	f.NamespacedTest("012-https-fallback-certificate", func(namespace string) {
+		Context("with fallback certificate", func() {
+			BeforeEach(func() {
+				contourConfig.TLS = config.TLSParameters{
+					FallbackCertificate: config.NamespacedName{
+						Name:      "fallback-cert",
+						Namespace: namespace,
+					},
+				}
+				f.Certs.CreateSelfSignedCert(namespace, "fallback-cert", "fallback-cert", "fallback.projectcontour.io")
+			})
+
+			testHTTPSFallbackCertificate(namespace)
+		})
 	})
-	It("020-global-rate-limiting-route-tls", func() {
-		testGlobalRateLimitingRouteTLS(f)
+
+	f.NamespacedTest("014-external-auth", testExternalAuth)
+
+	f.NamespacedTest("015-http-health-checks", testHTTPHealthChecks)
+
+	f.NamespacedTest("016-dynamic-headers", testDynamicHeaders)
+
+	f.NamespacedTest("017-host-header-rewrite", testHostHeaderRewrite)
+
+	f.NamespacedTest("018-external-name-service-insecure", testExternalNameServiceInsecure)
+
+	f.NamespacedTest("018-external-name-service-tls", testExternalNameServiceTLS)
+
+	f.NamespacedTest("019-local-rate-limiting-vhost", testLocalRateLimitingVirtualHost)
+
+	f.NamespacedTest("019-local-rate-limiting-route", testLocalRateLimitingRoute)
+
+	Context("global rate limiting", func() {
+		withRateLimitService := func(body e2e.NamespacedTestBody) e2e.NamespacedTestBody {
+			return func(namespace string) {
+				Context("with rate limit service", func() {
+					BeforeEach(func() {
+						contourConfig.RateLimitService = config.RateLimitService{
+							ExtensionService: fmt.Sprintf("%s/%s", namespace, f.Deployment.RateLimitExtensionService.Name),
+							Domain:           "contour",
+							FailOpen:         false,
+						}
+						require.NoError(f.T(),
+							f.Deployment.EnsureRateLimitResources(
+								namespace,
+								`
+domain: contour
+descriptors:
+  - key: generic_key
+    value: vhostlimit
+    rate_limit:
+      unit: hour
+      requests_per_unit: 1
+  - key: route_limit_key
+    value: routelimit
+    rate_limit:
+      unit: hour
+      requests_per_unit: 1
+  - key: generic_key
+    value: tlsvhostlimit
+    rate_limit:
+      unit: hour
+      requests_per_unit: 1
+  - key: generic_key
+    value: tlsroutelimit
+    rate_limit:
+      unit: hour
+      requests_per_unit: 1`))
```

Comment:
we could split this up since each descriptor here is relevant for one of the subsequent tests, but not urgent, can do as a separate cleanup, no need to add more to this PR :)